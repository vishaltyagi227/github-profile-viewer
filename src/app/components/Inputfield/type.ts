export interface IField {
  type: string;
  label: string;
  value: string;
  name: string;
  error?: boolean;
  helperText?: string;
  handleChange: (event: React.ChangeEvent<HTMLInputElement>) => void;
}
