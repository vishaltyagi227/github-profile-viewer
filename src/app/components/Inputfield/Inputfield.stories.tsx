import React from 'react';
import { Story, Meta } from '@storybook/react';
import InputField from '@components/Inputfield';
import { IField } from '@components/Inputfield/type';

export default {
  title: 'Components/InputField',
  component: InputField,
  argTypes: {
    value: { control: { disable: true } },
  },
} as Meta;

const Template: Story<IField> = ({ ...args }) => <InputField {...args} />;

export const Basic = Template.bind({});

Basic.args = {
  error: false,
  type: 'text',
  label: 'Name',
  name: 'userName',
  helperText: 'Please enter your name',
};

export const Validation = Template.bind({});

Validation.args = {
  ...Basic.args,
  error: true,
  helperText: 'Name is not valid',
};
