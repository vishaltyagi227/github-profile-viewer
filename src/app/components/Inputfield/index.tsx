import React from 'react';
import { TextField } from '@mui/material';

import { VARIANT } from '@constant/index';
import useStyles from './style';
import { IField } from './type';

const Inputfield = ({
  type,
  label,
  value,
  name,
  error,
  helperText,
  handleChange,
}: IField): JSX.Element => {
  const classes = useStyles();

  return (
    <TextField
      error={error}
      type={type}
      variant={VARIANT.outlined}
      label={label}
      className={classes.input}
      value={value}
      name={name}
      onChange={handleChange}
      helperText={helperText}
    />
  );
};

Inputfield.defaultProps = {
  error: false,
  helperText: '',
};

export default Inputfield;
