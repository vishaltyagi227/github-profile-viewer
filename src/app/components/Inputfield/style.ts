import { Theme } from '@mui/material/styles';
import { createStyles, makeStyles } from '@mui/styles';

const useStyles = makeStyles((theme: Theme) =>
  createStyles({
    input: {
      marginBlock: theme.spacing(3),
      width: '100%',
    },
  })
);

export default useStyles;
