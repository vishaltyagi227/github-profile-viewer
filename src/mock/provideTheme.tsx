import React, { ReactElement } from 'react';
import { ThemeProvider } from '@mui/material/styles';
import theme from '@theme/index';

export default (ui: ReactElement): ReactElement => (
  <ThemeProvider theme={theme}>{ui}</ThemeProvider>
);
