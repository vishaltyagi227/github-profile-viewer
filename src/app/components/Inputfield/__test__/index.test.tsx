import React, { useState } from 'react';
import { render, cleanup, RenderResult } from '@testing-library/react';
import renderer from 'react-test-renderer';
import provideTheme from '@mock/provideTheme';
import userEvent from '@testing-library/user-event';
import Inputfield from '../index';

const MockWrapper = () => {
  const [value, setValue] = useState('');
  const handleChangeMock = (e: React.ChangeEvent<HTMLInputElement>) => {
    setValue(e.target.value);
  };
  return provideTheme(
    <Inputfield
      label="input Field"
      type="text"
      name="text"
      handleChange={handleChangeMock}
      value={value}
    />
  );
};

describe('<InputField />', () => {
  let container: RenderResult;

  beforeEach(() => {
    container = render(<MockWrapper />);
  });

  afterEach(cleanup);

  it('Should render correctly', () => {
    expect(container.getByRole('textbox')).toBeInTheDocument();
  });

  it('Should have value as user type', () => {
    userEvent.type(container.getByRole('textbox'), 'username');
    expect(container.getByRole('textbox')).toHaveValue('username');
  });
});

describe('<InputField /> Snapshot', () => {
  afterEach(cleanup);

  it('Should match with snapshot', () => {
    const tree = renderer.create(<MockWrapper />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
