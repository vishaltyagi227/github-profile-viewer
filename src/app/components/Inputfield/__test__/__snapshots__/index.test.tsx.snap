// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<InputField /> Snapshot Should match with snapshot 1`] = `
<div
  className="MuiFormControl-root MuiTextField-root makeStyles-input-3 css-1u3bzj6-MuiFormControl-root-MuiTextField-root"
>
  <label
    className="MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined MuiFormLabel-root MuiFormLabel-colorPrimary css-14s5rfu-MuiFormLabel-root-MuiInputLabel-root"
    data-shrink={false}
  >
    input Field
  </label>
  <div
    className="MuiOutlinedInput-root MuiInputBase-root MuiInputBase-colorPrimary MuiInputBase-formControl css-9ddj71-MuiInputBase-root-MuiOutlinedInput-root"
    onClick={[Function]}
  >
    <input
      aria-invalid={false}
      autoFocus={false}
      className="MuiOutlinedInput-input MuiInputBase-input css-1t8l2tu-MuiInputBase-input-MuiOutlinedInput-input"
      disabled={false}
      name="text"
      onAnimationStart={[Function]}
      onBlur={[Function]}
      onChange={[Function]}
      onFocus={[Function]}
      required={false}
      type="text"
      value=""
    />
    <fieldset
      aria-hidden={true}
      className="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
    >
      <legend
        className="css-dhh0px"
      >
        <span>
          input Field
        </span>
      </legend>
    </fieldset>
  </div>
  
</div>
`;
